### DISCUSSION TITLE
Questions about the use of getStateForAction

### DISCUSSION THREAD

**Discussion by jswangtao** [react-navigation/react-navigation#10965] [Q&A]
### Current behavior

In 4.x, I can use the following code to quickly implement routing customization and interception operations
```
const Navigator = createStackNavigator()
Navigator.router.getStateForAction = (action, state) => {
console.log(action)
return defaultGetStateForAction(action, state)
}
```

### Expected behavior

Now that I upgrade to 6.x, how do I implement the same routing customization and interception operations? The one on the official website can't understand how to use it. Is there a complete example?

### Reproduction

h

### Platform

- [ ] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/​bottom-tabs
- [ ] @react-navigation/​drawer
- [ ] @react-navigation/​material-bottom-tabs
- [ ] @react-navigation/​material-top-tabs
- [ ] @react-navigation/​stack
- [ ] @react-navigation/​native-stack

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |
| @react-navigation/bottom-tabs          |
| @react-navigation/drawer               |
| @react-navigation/material-bottom-tabs |
| @react-navigation/material-top-tabs    |
| @react-navigation/stack                |
| @react-navigation/native-stack         |
| react-native-safe-area-context         |
| react-native-screens                   |
| react-native-gesture-handler           |
| react-native-reanimated                |
| react-native-tab-view                  |
| react-native-pager-view                |
| react-native                           |
| expo                                   |
| node                                   |
| npm or yarn                            |

**Comment by victormak** [comment]
I hope there is an official example of it

**Reply by Samykills** [reply]
@satya164 is there any way we can do this with react-nav6+?

**Reply by satya164** [reply]
https://reactnavigation.org/docs/custom-navigators#extending-navigators
